### PR TITLE
Swagger Documentation for Clients API

### PR DESCRIPTION
**Pull request by luckyman20**
@Sanyam96 Please review it ASAP !!

### CODE REVIEW

**Review by Sanyam96 — CHANGES_REQUESTED** [review]
ClientAddressApiResourceSwagger.java
the name of this file is not correct

Please rename it to ClientAddressApiResourcesSwagger.java

**Comment by luckyman20** [comment]
@Sanyam96 Checked the Swagger UI

File: fineract-provider/src/main/java/org/apache/fineract/portfolio/client/api/ClientsApiResource.java [hunk 1]
```diff
@@ -127,16 +115,15 @@ public String retrieveTemplate(@Context final UriInfo uriInfo, @ApiParam(value =
...
-            @QueryParam("officeId") final Long officeId, @QueryParam("externalId") final String externalId,
-            @QueryParam("displayName") final String displayName, @QueryParam("firstName") final String firstname,
-            @QueryParam("lastName") final String lastname, @QueryParam("underHierarchy") final String hierarchy,
-            @QueryParam("offset") final Integer offset, @QueryParam("limit") final Integer limit,
-            @QueryParam("orderBy") final String orderBy, @QueryParam("sortOrder") final String sortOrder,
-            @QueryParam("orphansOnly") final Boolean orphansOnly) {
+    @ApiOperation(value = "List Clients", notes = "The list capability of clients can support pagination and sorting.\n\n" + "Example Requests:\n" + "\n" + "clients\n" + "\n" + "clients?fields=displayName,officeName,timeline\n" + "\n" + "clients?offset=10&limit=50\n" + "\n" + "clients?orderBy=displayName&sortOrder=DESC" )
+    @ApiResponses({@ApiResponse(code = 200, message = "OK", response = ClientsApiResourceSwagger.GetClientsResponse.class, responseContainer = "List")})
```
Comment (Sanyam96) [inline fineract-provider/src/main/java/org/apache/fineract/portfolio/client/api/ClientsApiResource.java]: Response is not correct

File: fineract-provider/src/main/java/org/apache/fineract/portfolio/client/api/ClientTransactionsApiResource.java [hunk 2]
```diff
@@ -77,9 +69,9 @@ public ClientTransactionsApiResource(final PlatformSecurityContext context,
     @Consumes({ MediaType.APPLICATION_JSON })
     @Produces({ MediaType.APPLICATION_JSON })
     @ApiOperation(value = "List Client Transactions", notes = "The list capability of client transaction can support pagination." + "\n\nMandatory Arguments : " + "offset\n" + "    Integer Mandatory, defaults to 0 \n" + "    Indicates the result from which pagination starts\n" + "limit\n" + "    Integer Mandatory, defaults to 200 \n" + "    Restricts the size of results returned. To override the default and return all entries you must explicitly pass a non-positive integer value for limit e.g. limit=0, or limit=-1\n" + "\n" + "\n\n" + "Example Requests:\n" + "clients/189/transactions\n" + "clients/189/transactions?offset=10&limit=50" )
```
Comment (Sanyam96) [inline fineract-provider/src/main/java/org/apache/fineract/portfolio/client/api/ClientTransactionsApiResource.java]: New Line character between 2 different Example Request is missing '\n'

File: fineract-provider/src/main/java/org/apache/fineract/portfolio/client/api/ClientTransactionsApiResource.java [hunk 3]
```diff
@@ -114,11 +106,10 @@ public String retrieveClientTransaction(@PathParam("clientId") final Long client
...
-    @ApiOperation(value = "Undo a Client Transaction")
-    @ApiImplicitParams({@ApiImplicitParam(value = "body", dataType = "body", dataTypeClass = ClientData.class)})
-    @ApiResponses({@ApiResponse(code = 200, message = "", response = CommandProcessingResult.class)})
-    public String undoClientTransaction(@PathParam("clientId") final Long clientId, @PathParam("transactionId") final Long transactionId,
-            @QueryParam("command") final String commandParam, @ApiParam(hidden = true) final String apiRequestBodyAsJson) {
+    @ApiOperation(value = "Undo a Client Transaction", notes = "Undoes a Client Transaction")
+    @ApiResponses({@ApiResponse(code = 200, message = "OK", response = ClientTransactionsApiResourceSwagger.PostClientsClientIdTransactionsTransactionIdResponse.class)})
+    public String undoClientTransaction(@PathParam("clientId") @ApiParam(value = "clientId") final Long clientId, @PathParam("transactionId") @ApiParam(value = "transactionId") final Long transactionId,
+            @QueryParam("command") @ApiParam(example = "command") final String commandParam, @ApiParam(hidden = true) final String apiRequestBodyAsJson) {
```
Comment (Sanyam96) [inline fineract-provider/src/main/java/org/apache/fineract/portfolio/client/api/ClientTransactionsApiResource.java]: Please correct the annotation to
@ApiParam(value = "command")

**Review by Sanyam96 — CHANGES_REQUESTED** [review]

**Comment by Sanyam96** [comment]
Rest all looks good
Please modify these changes asap ;)

File: fineract-provider/src/main/java/org/apache/fineract/portfolio/client/api/ClientsApiResource.java (hunk 1 above)
Comment (luckyman20) [reply fineract-provider/src/main/java/org/apache/fineract/portfolio/client/api/ClientsApiResource.java]: Already solved

**Comment by luckyman20** [comment]
@Sanyam96 Please review it now !! I have made the changes

File: fineract-provider/src/main/java/org/apache/fineract/portfolio/client/api/ClientsApiResourceSwagger.java [hunk 4]
```diff
@@ -0,0 +1,464 @@
...
+            public String displayName;
+            @ApiModelProperty(example = "1")
+            public Integer officeId;
+            @ApiModelProperty(example = "Head Office")
+            public String officeName;
+        }
+
+        @ApiModelProperty(example = "2")
+        public Integer totalFilteredRecords;
+        public GetClientsPageItemsResponse pageItems;
```
Comment (Sanyam96) [inline fineract-provider/src/main/java/org/apache/fineract/portfolio/client/api/ClientsApiResourceSwagger.java]: this should be Set

File: fineract-provider/src/main/java/org/apache/fineract/portfolio/client/api/ClientsApiResource.java (hunk 1 above)
Comment (Sanyam96) [reply fineract-provider/src/main/java/org/apache/fineract/portfolio/client/api/ClientsApiResource.java]: the response type of pageItems is a list or set
please check with the current swagger UI and original docs

**Review by Sanyam96 — CHANGES_REQUESTED** [review]

**Review by Sanyam96 — APPROVED** [review]

**Pull request merged**